packages/nimble-components/src/table/index.ts

```diff
@@ -336,35 +373,103 @@ export class Table<
             return { ...record };
         });
         this.validateWithData(data);
+        this.updateTableOptions(
+            {
+                data,
+                state: {
+                    expanded: true
+                }
+            },
+            true
+        );
+        const expandedState = this.getInflatedExpandedState() as {
+            [key: string]: boolean
+        };
+        for (const collapsedRowId of Object.keys(this.collapsedState)) {
+            if (this.collapsedState[collapsedRowId]) {
+                expandedState[collapsedRowId] = false;
+            }
+        }
         this.updateTableOptions({
-            data
+            state: {
+                expanded: expandedState
+            }
         });
     }
 
+    private getInflatedExpandedState(): TanStackExpandedState {
+        if (this.table.options.state.expanded === true) {
+            const expandedState = {} as { [key: string]: boolean };
+            for (const row of this.table.getRowModel().rows) {
+                expandedState[row.id] = true;
+            }
+            return expandedState;
+        }
+
+        return this.table.options.state.expanded!;
+    }
+
     private refreshRows(): void {
         const rows = this.table.getRowModel().rows;
-        this.tableData = rows.map(row => {
+        this.tableData = rows.map((row, i) => {
+            const groupColumn = this.getGroupRowColumn(i);
```

Comment:
Do you need this line, or the `getGroupRowColumn()` private function? I only see the value in one place, and I think it could be re-written as:

```ts
groupRowValue: row.getIsGrouped()
    ? get-the-group-value
    : undefined
```